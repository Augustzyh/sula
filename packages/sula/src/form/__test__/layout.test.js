import React from 'react';
import renderer from 'react-test-renderer';
import { Form } from '..';
import '../../__tests__/common';

describe('layout', () => {
  it('itemlayout', () => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: true,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
    const form = (
      <Form
        itemLayout={{
          cols: {
            xxl: 4,
            xl: 3,
            lg: 2,
            md: 2,
            sm: 1,
            xs: 1,
          },
        }}
        fields={[
          {
            name: 'input1',
            label: 'input1',
            field: 'input',
          },
          {
            name: 'input2',
            label: 'input2',
            field: 'input',
          },
          {
            name: 'input3',
            label: 'input3',
            field: 'input',
            initialVisible: false,
          },
          {
            name: 'input4',
            label: 'input4',
            field: 'input',
          },
          {
            name: 'input5',
            label: 'input5',
            field: 'input',
          },
          {
            name: 'input6',
            label: 'input6',
            field: 'input',
          },
        ]}
      />
    );

    expect(renderer.create(form).toJSON()).toMatchSnapshot();
  });

  it('item span 24', () => {
    const form = (
      <Form
        itemLayout={{
          span: 24,
        }}
        fields={[
          {
            name: 'input1',
            label: 'input1',
            field: 'input',
          },
          {
            name: 'input2',
            label: 'input2',
            field: 'input',
          },
          {
            name: 'input3',
            label: 'input3',
            field: 'input',
          },
          {
            name: 'group',
            fields: [
              {
                name: 'input4',
                label: 'input4',
                field: 'input',
              },
            ],
          },
          {
            name: 'group2',
            fields: [
              {
                name: 'input5',
                label: 'input5',
                field: 'input',
              },
              {
                name: 'input6',
                label: 'input6',
                field: 'input',
              },
            ],
          },
        ]}
      />
    );

    expect(renderer.create(form).toJSON()).toMatchSnapshot();
  });
});
