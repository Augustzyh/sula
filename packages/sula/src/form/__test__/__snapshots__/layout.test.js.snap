// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout item span 24 1`] = `
<div
  className="ant-spin-nested-loading"
>
  <div
    className="ant-spin-container"
  >
    <form
      className="ant-form ant-form-horizontal"
      onSubmit={[Function]}
    >
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-8 ant-form-item-label"
          style={Object {}}
        >
          <label
            className=""
            htmlFor="input1"
            title="input1"
          >
            input1
          </label>
        </div>
        <div
          className="ant-col ant-col-12 ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                className="ant-input"
                id="input1"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-8 ant-form-item-label"
          style={Object {}}
        >
          <label
            className=""
            htmlFor="input2"
            title="input2"
          >
            input2
          </label>
        </div>
        <div
          className="ant-col ant-col-12 ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                className="ant-input"
                id="input2"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-8 ant-form-item-label"
          style={Object {}}
        >
          <label
            className=""
            htmlFor="input3"
            title="input3"
          >
            input3
          </label>
        </div>
        <div
          className="ant-col ant-col-12 ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                className="ant-input"
                id="input3"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-8 ant-form-item-label"
          style={Object {}}
        >
          <label
            className=""
            htmlFor="input4"
            title="input4"
          >
            input4
          </label>
        </div>
        <div
          className="ant-col ant-col-12 ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                className="ant-input"
                id="input4"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-8 ant-form-item-label"
          style={Object {}}
        >
          <label
            className=""
            htmlFor="input5"
            title="input5"
          >
            input5
          </label>
        </div>
        <div
          className="ant-col ant-col-12 ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                className="ant-input"
                id="input5"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-8 ant-form-item-label"
          style={Object {}}
        >
          <label
            className=""
            htmlFor="input6"
            title="input6"
          >
            input6
          </label>
        </div>
        <div
          className="ant-col ant-col-12 ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                className="ant-input"
                id="input6"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`layout itemlayout 1`] = `
<div
  className="ant-spin-nested-loading"
>
  <div
    className="ant-spin-container"
  >
    <form
      className="ant-form ant-form-horizontal"
      onSubmit={[Function]}
    >
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-8 ant-form-item-label"
          style={Object {}}
        >
          <label
            className=""
            htmlFor="input1"
            title="input1"
          >
            input1
          </label>
        </div>
        <div
          className="ant-col ant-col-12 ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                className="ant-input"
                id="input1"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-8 ant-form-item-label"
          style={Object {}}
        >
          <label
            className=""
            htmlFor="input2"
            title="input2"
          >
            input2
          </label>
        </div>
        <div
          className="ant-col ant-col-12 ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                className="ant-input"
                id="input2"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-8 ant-form-item-label"
          style={Object {}}
        >
          <label
            className=""
            htmlFor="input3"
            title="input3"
          >
            input3
          </label>
        </div>
        <div
          className="ant-col ant-col-12 ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                className="ant-input"
                id="input3"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-8 ant-form-item-label"
          style={Object {}}
        >
          <label
            className=""
            htmlFor="input4"
            title="input4"
          >
            input4
          </label>
        </div>
        <div
          className="ant-col ant-col-12 ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                className="ant-input"
                id="input4"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-8 ant-form-item-label"
          style={Object {}}
        >
          <label
            className=""
            htmlFor="input5"
            title="input5"
          >
            input5
          </label>
        </div>
        <div
          className="ant-col ant-col-12 ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                className="ant-input"
                id="input5"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-8 ant-form-item-label"
          style={Object {}}
        >
          <label
            className=""
            htmlFor="input6"
            title="input6"
          >
            input6
          </label>
        </div>
        <div
          className="ant-col ant-col-12 ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                className="ant-input"
                id="input6"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
